reframe/frontend/cli.py

```diff
@@ -229,6 +238,20 @@ def main():
     # Init modules system
     init_modules_system(system.modules_system)
 
+    if options.module_map_file:
+        try:
+            get_modules_system().load_mapping_from_file(
+                options.module_map_file)
+        except Exception as e:
+                raise ConfigError('could not set module mapping: %s' % e)
```

Comment:
Here you are at the top-level of the code. If you raise any exception, it will fail badly. You should do the following:

```python
try:
   # Here you can process both options; no need for two try/except
except ConfigError, OSError as e:
    printer.error('could not load module mappings: %s' % e)
    sys.exit(1)
```

Reply:
Here, you should better check for a `ReframeError` rather than `ConfigError`. You should remember to remove it also from the `import` above.

Reply:
Isn't it necessary one `try/except` for `if options.module_map_file` and other for `if options.module_mappings`?

Reply:
It's not bad how you have it and it is generally a good practice not to have very large try/except blocks. But here both options are eventually calling the same methods, so I see it more as redundant code.